### DISCUSSION TITLE
Rye with `uv`

### DISCUSSION THREAD

**Discussion by noahlias** [astral-sh/rye#701] [Q&A]
I have anaconda installed locally now, but I enabled the latest feature `uv` and after that I get the following error when `rye list`, 

![image](https://github.com/mitsuhiko/rye/assets/3949397/85e87f9f-50e0-4624-bd44-a63cdfd88487)

I know it has a workaround,just run
```bash
unset CONDA_PREFIX
```
 is there any other way to fix this?

**Comment by garam-kim1** [comment]
Additionally, I'm curious if there's a method to install libraries directly into a Conda environment instead of using a virtual environment (venv). Could I get some guidance or recommend best practices for this?

**Reply by stimw** [reply]
When creating an environment with conda, it automatically includes Python and some packages, which can cause confusion. 

Moreover, if there are many projects, it becomes difficult to recognize which conda environment a particular project belongs to. 

Rye addresses this issue. If you still want to use conda no matter what, you might try **pdm**. 

Currently, I only use conda for deep learning.

**Reply by garam-kim1** [reply]
Thank you for the information!

Is there an efficient way to manage multiple virtual environments within a single project? For instance, when I need to test my code with a new version of Python (or a new version of a library), I usually create a new virtual environment for that specific Python version for testing purposes. I primarily use Python for deep learning. Perhaps, as you mentioned, Rye might not be suitable for my work?

**Reply by stimw** [reply]
Deep learning frameworks require many dependencies unrelated to Python, such as CUDA. This is where conda excels, as it resolves system-level dependencies for you. In my opinion, conda is the best choice for deep learning. Rye is more suitable for Python engineering projects, such as backend development.

**Comment by stimw** [comment]
I believe that preventing conda from automatically activating the base environment is best practice.

Only use conda for deep learning like pytorch and cuda.
